src/main/java/hudson/plugins/analysis/core/BuildHistory.java

```diff
@@ -187,11 +190,27 @@ private boolean isSuccessfulAction(final ResultAction<? extends BuildResult> act
         return null;
     }
 
-    private boolean hasValidResult(final AbstractBuild<?, ?> build) {
+    /**
+     * Added for backward compatibility. It generates <pre>AbstractBuild getReferenceBuild()</pre> bytecode during the build
+     * process, so old implementations can use that signature.
+     * 
+     * @see {@link WithBridgeMethods}
+     */
+    @Deprecated
+    private Object getReferenceAbstractBuild(Run run, Class targetClass) {
+        if (run instanceof AbstractBuild) {
+            return (AbstractBuild) run;
+        }
+        else {
+            return null;
+        }
+    }
+
```

Comment:
The other adapter methods use 

```
run instanceof AbstractBuild ? (AbstractBuild) run : null;
``` 
It would make sense to always use the same pattern.